admin/aws.py

```diff
@@ -0,0 +1,203 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Effectful interface to boto.
+"""
+
+from characteristic import attributes, Attribute
+from effect import sync_performer, TypeDispatcher
+
+import boto
+
+
+@attributes([
+    "bucket",
+    "prefix",
+    "target_prefix",
+])
+class UpdateS3RoutingRule(object):
+    """
+    Update a routing rule for an S3 bucket website endpoint to point to a new
+    path.
+
+    If the path is changed, return the old path.
+
+    :ivar bytes bucket: Name of bucket to change routing rule for.
+    :ivar bytes prefix: Prefix to change routing rule for.
+    :ivar bytes target_prefix: Target prefix to redirect to.
+    """
+
+
+@sync_performer
+def perform_update_s3_routing_rule(dispatcher, intent):
+    s3 = boto.connect_s3()
+    bucket = s3.get_bucket(intent.bucket)
+    config = bucket.get_website_configuration_obj()
+    rule = [rule for rule in config.routing_rules if
+            rule.condition.key_prefix == intent.prefix][0]
+    if rule.redirect.replace_key_prefix == intent.target_prefix:
+        return None
+    else:
+        old_prefix = rule.redirect.replace_key_prefix
+        rule.redirect.replace_key_prefix = intent.target_prefix
+        bucket.set_website_configuration()
```

Comment:
We think that this is just `bucket.set_website_configuration()` > `bucket.set_website_configuration(config)` and in testing this superficially works.